import-export-cli/impl/importAPI.go

```diff
@@ -950,3 +494,129 @@ func ImportAPI(accessOAuthToken, adminEndpoint, importEnvironment, importPath, a
+//directory
+func envParamsDirectoryProcess(importPath, paramsPath, importEnvironment string) error {
+	apiParams, err := params.LoadApiParamsFromDirectory(paramsPath)
+	if err != nil {
+		return err
+	}
+	// check whether import environment is included in api configuration
+	envParams := apiParams.GetEnv(importEnvironment)
+	if envParams == nil {
+		utils.Logln(utils.LogPrefixInfo + "Using default values as the environment is not present in api_param.yaml file")
+	} else {
+		//If environment parameters are present in parameter file
+		err = handleEnvParams(importPath, envParams)
+		if err != nil {
+			return err
+		}
+
+		//copy all the content in the params directory into the artifact to be imported
+		utils.CopyDirectoryContents(paramsPath, importPath)
+	}
+	return nil
+}
+
+// handleCustomizedParameters handles the configurations provided with apiParams file and the resources that needs to
+// transfer to server side will bundle with the artifact to be imported.
+func handleCustomizedParameters(importPath, paramsPath, importEnvironment string) error {
+	utils.Logln(utils.LogPrefixInfo+"Loading parameters from", paramsPath)
+	if strings.Contains(paramsPath, ".yaml") {
+		utils.Logln(utils.LogPrefixInfo+"Processing Params file", paramsPath)
+		err := envParamsFileProcess(importPath, paramsPath, importEnvironment)
+		if err != nil {
+			return err
+		}
+	} else {
+		utils.Logln(utils.LogPrefixInfo+"Processing Params directory", paramsPath)
+		err := envParamsDirectoryProcess(importPath, paramsPath, importEnvironment)
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+//Process env params and create a temp env_parmas.yaml in temp artifact
+func handleEnvParams(apiDirectory string, environmentParams *params.Environment) error {
+	// read api from Meta-information
+	apiPath := filepath.Join(apiDirectory, "Meta-information", "api")
+	utils.Logln(utils.LogPrefixInfo + "Reading API definition: ")
+	fp, jsonContent, err := resolveYamlOrJSON(apiPath)
+	if err != nil {
+		return err
+	}
+	utils.Logln(utils.LogPrefixInfo+"Loaded definition from:", fp)
+	api, err := gabs.ParseJSON(jsonContent)
+	if err != nil {
+		return err
+	}
+
+	envParamsJson, err := jsoniter.Marshal(environmentParams.Config)
+	if err != nil {
+		return err
+	}
+
+	apiPath = filepath.Join(apiDirectory, "Meta-information", "api.yaml")
+	var apiParamsPath string
+	utils.Logln(utils.LogPrefixInfo+"Writing merged API to:", apiPath)
+
+	// write this to disk
+	content, err := utils.JsonToYaml(api.Bytes())
+	if err != nil {
+		return err
+	}
+
+	apiParams, err := gabs.ParseJSON(envParamsJson)
+	paramsContent, err := utils.JsonToYaml(apiParams.Bytes())
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(apiPath, content, 0644)
+
+	//over-write the api_params.file with latest configurations
+	apiParamsPath = filepath.Join(apiDirectory, "api_params.yaml")
+	err = ioutil.WriteFile(apiParamsPath, paramsContent, 0644)
```

Comment:
In the review meeting we discussed that we were going to keep the project zip unchanged and were simply going to include the env param file as part of the intermediate zip file as follows,

```
intermediate-zip
        |
        |____ project-zip
        |____ params
                 |_____ param.yaml
```
But seems with this code we are modifying the project zip file to include the param yaml inside it.

Reply:
This change will be a part of the next feature I am working on, at the moment. Will incorporate this change in that separate PR. Thanks very much for pointing this out.